### ISSUE TITLE
Using Docker for NEST CI

### ISSUE THREAD

**Issue by lekshmideepu** [nest/nest-simulator#641]
Our builds on Travis takes a lot of time for different reasons like:
   - installing all the necessary packages on an "empty" worker machine 
   - running all possible combinations of the jobs (job matrix).
 Also in future we would like to test with multiple python versions and multiple OS's which would again take a lot of time.
 
Since Travis supports docker images, I think it is a good idea to have a NEST docker image which also helps us to speed up our Travis build to some extent.

**Comment by heplesser** [comment]
@lekshmideepu This sounds like a very fine idea. Could you explore this a bit further? Would this also allow us to move to a newer Ubuntu version and recent compiler and tools? The pep8 checker we currently use on travis is quite old and leads sometimes to results inconsistent with newer versions.

**Comment by lekshmideepu** [comment]
@heplesser Sure, I could explore the various options.

**Comment by heplesser** [comment]
@lekshmideepu I was wondering about the status of this project? I just noticed that Travis builds hang because the llvm.org website is currently unreachable, so clang-format cannot be downloaded. It would be really great if we could avoid such "live" dependencies in the future.

**Comment by heplesser** [comment]
@lekshmideepu  Any news? We see more frequently now that jobs stop due to timeouts, so it would be good to move forward.

**Comment by heplesser** [comment]
@lekshmideepu Ping!

**Comment by lekshmideepu** [comment]
During the last hackathon, I had a discussion with @steffengraber  With the current docker image of NEST, it would be difficult to run it on Travis. Plan is to work together with Steffen once he is back from vacation.

**Comment by lekshmideepu** [comment]
Together with @steffengraber we did a "proof of concept" [build](https://travis-ci.org/lekshmideepu/nest-simulator/jobs/545710474#L5299) for the NEST docker build on Travis. The error showed is not related to the docker file as such.

**Comment by lekshmideepu** [comment]
We  (@steffengraber) are working on the NEST build docker image with all the build dependencies. Once this is ready, planning to test this on Travis and check if it brings any advantages like saving build time and new possibilities that we don't know right now....

**Comment by heplesser** [comment]
@lekshmideepu Is this still relevant now that we have moved to Github actions?

**Comment by heplesser** [comment]
Moved to Github actions, no longer relevant.
